### PR TITLE
ddl, owner: set owner op with lease

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/43850

Problem Summary:
There a already expired additional owner,  resulting in no normal DDL owner in the cluster..
```
ETCDCTL_API=3 etcdctl --endpoints=127.0.0.1:2379 lease timetolive 2231882562b6cb5f
{"level":"warn","ts":"2023-05-17T00:40:56.476894+0800","caller":"flags/flag.go:93","msg":"unrecognized environment variable","environment-variable":"ETCDCTL_API=3"}
lease 2231882562b6cb5f already expired
```

### What is changed and how it works?
* Set the value to etcd with lease.
* Clean up the log(get global state and global state change) when the TiDB is notOwner.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

1. Using tiup start 2 TiDBs(master brach) cluster.
2. Kill a TiDB and start(before start do `update mysql.tidb set VARIABLE_VALUE = 143 WHERE VARIABLE_NAME="tidb_server_version";`) it with `--with-mock-upgrade=true` and update `var SupportUpgradeStateVer = 
version142`
3. Kill another TiDB, and start it.
Before this PR we have an owner key with already expired. After this PR, doesn't  no additional owner.
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: ddl/job_table.go [hunk 1]
```diff
@@ -266,22 +266,21 @@ func (d *ddl) startDispatchLoop() {
...
-			continue
-		}
-		isFirst = false
 		if !d.isOwner() {
 			isOnce = true
 			d.once.Store(true)
 			time.Sleep(dispatchLoopWaitingDuration)
 			continue
 		}
+		if err := d.needCheckClusterState(isOnce); err != nil {
```
Comment (Defined2014) [inline ddl/job_table.go:280]: Why move to here? Don't want to `checkClusterState` in first time? And maybe we could remove line 301.

File: owner/manager.go [hunk 2]
```diff
@@ -392,9 +397,10 @@ func (m *ownerManager) SetOwnerOpValue(ctx context.Context, op OpType) error {
 		}
 	})
 
+	leaseOp := clientv3.WithLease(clientv3.LeaseID(m.sessionLease.Load()))
 	resp, err := m.etcdCli.Txn(ctx).
 		If(clientv3.Compare(clientv3.ModRevision(ownerKey), "=", modRevision)).
-		Then(clientv3.OpPut(ownerKey, string(newOwnerVal))).
+		Then(clientv3.OpPut(ownerKey, string(newOwnerVal), leaseOp)).
```
Comment (Defined2014) [inline owner/manager.go:403]: when will the owner renew lease?

File: ddl/job_table.go (hunk 1 above)
Comment (zimulala) [reply ddl/job_table.go:280]: `checkClusterState` is first time in `serverStateSyncer.Init`.

File: owner/manager.go (hunk 2 above)
Comment (zimulala) [reply owner/manager.go:403]: Here https://github.com/pingcap/tidb/pull/43894/files#diff-a0eb33bb4d5cbd8ab32a999bdabe4e3c35dba3e9ac5ad03fa3b8d033334ff8f7R250

Comment (Defined2014) [reply owner/manager.go:403]: Will this lead to frequent owner changes?

**Review by Defined2014 — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

File: tests/realtikvtest/sessiontest/BUILD.bazel [hunk 3]
```diff
@@ -2,7 +2,7 @@ load("@io_bazel_rules_go//go:def.bzl", "go_test")
 
 go_test(
     name = "sessiontest_test",
-    timeout = "moderate",
+    timeout = "long",
```
Comment (hawkingrei) [inline tests/realtikvtest/sessiontest/BUILD.bazel:5]: long indicates that this test can run for a maximum of 15 minutes. so after this PR, this test needs more time to complete. It sounds a bit odd.

**Comment by hawkingrei** [comment]
```
=== NAME  TestLoadRemote

    result.go:49: 

        	Error Trace:	testkit/result.go:49

        	            				tests/realtikvtest/loaddatatest/load_data_test.go:860

        	            				tests/realtikvtest/loaddatatest/load_data_test.go:837

        	Error:      	Not equal: 

        	            	expected: "[1 110 123]\n[2 220 123]\n[3 330 123]\n[4 440 123]\n[5 550 123]\n[6 660 123]\n[7 770 123]\n[8 880 123]\n[9 990 123]\n"

        	            	actual  : ""

        	            	

        	            	Diff:

        	            	--- Expected

        	            	+++ Actual

        	            	@@ -1,10 +1 @@

        	            	-[1 110 123]

        	            	-[2 220 123]

        	            	-[3 330 123]

        	            	-[4 440 123]

        	            	-[5 550 123]

        	            	-[6 660 123]

        	            	-[7 770 123]

        	            	-[8 880 123]

        	            	-[9 990 123]

        	            	 

        	Test:       	TestLoadRemote

        	Messages:   	sql:SELECT * FROM load_data.cols_and_vars;, args:[]

=== NAME  TestLoadRemote/TestColumnsAndUserVars

```

Here is a flaky test that is found in the log.

**Comment by Benjamin2037** [comment]
Can we let QA add these test case into their test cases system? otherwise we will loss our test case property.

**Pull request merged**